src/arch/xtensa/CMakeLists.txt

```diff
@@ -255,66 +241,11 @@ add_custom_target(sof_post_process
 # contains extra output that should be generated for bin target
 add_custom_target(bin_extras)
 
-# bootloader binary
-
-if(build_bootloader)
-	add_executable(bootloader "")
-
-	add_dependencies(bootloader rimage_ep)
-	# This line should be next to the boot_module
-	# (resp. base_module) definitions but there are too many of
-	# them; one per platform. So do it only once here instead.
-	add_dependencies(boot_module rimage_ep)
-	add_dependencies(base_module rimage_ep)
-
-	target_include_directories(bootloader   PRIVATE ${PROJECT_SOURCE_DIR}/rimage/src/include)
-	target_include_directories(boot_module  PRIVATE ${PROJECT_SOURCE_DIR}/rimage/src/include)
-	target_include_directories(base_module  PRIVATE ${PROJECT_SOURCE_DIR}/rimage/src/include)
-
-
-	target_link_libraries(bootloader PRIVATE sof_options)
-	add_local_sources(bootloader xtos/_vectors.S ${PROJECT_SOURCE_DIR}/src/platform/${family_path}/boot_entry.S ${PROJECT_SOURCE_DIR}/src/platform/${family_path}/boot_loader.c)
-	target_link_libraries(bootloader PRIVATE reset)
-	target_link_libraries(bootloader PRIVATE hal)
-	target_link_libraries(bootloader PRIVATE "-T${PROJECT_BINARY_DIR}/${platform_bootldr_ld_script}")
-	sof_add_ld_script(bootloader ${platform_bootldr_ld_script})
-	sof_append_relative_path_definitions(bootloader)
-
-	add_custom_target(
-		bootloader_dump
-		COMMAND ${CMAKE_COMMAND} -E copy bootloader bootloader-${fw_name}
-		COMMAND ${CMAKE_OBJCOPY} -O binary -j .data ${PROJECT_BINARY_DIR}/src/platform/${platform_folder}/boot_module mod-boot-${fw_name}.bin
-		COMMAND ${CMAKE_OBJCOPY} --add-section .module=mod-boot-${fw_name}.bin --set-section-flags .module=load,readonly bootloader-${fw_name}
-		COMMAND ${CMAKE_OBJCOPY} -O binary bootloader bootloader-${fw_name}.bin
-		COMMAND ${CMAKE_OBJDUMP} -h -D bootloader > bootloader-${fw_name}.lmap
-		COMMAND ${CMAKE_OBJDUMP} -S bootloader > bootloader-${fw_name}.lst
-		COMMAND ${CMAKE_OBJDUMP} -D bootloader > bootloader-${fw_name}.dis
-		DEPENDS bootloader boot_module
-		VERBATIM
-		USES_TERMINAL
-	)
-
-	set(bootloader_binary_path bootloader-${fw_name})
-
-	# Add 'base_module' section to sof ELF
-	add_custom_target(
-		process_base_module
-		COMMAND ${CMAKE_OBJCOPY} -O binary -j .data ${PROJECT_BINARY_DIR}/src/platform/${platform_folder}/base_module mod-${fw_name}.bin
-		COMMAND ${CMAKE_OBJCOPY} --add-section .module=mod-${fw_name}.bin --set-section-flags .module=load,readonly sof-pre sof-${fw_name}
-		DEPENDS prepare_sof_post_process base_module bootloader_dump
-		VERBATIM
-		USES_TERMINAL
-	)
-
-else()
-	set(bootloader_binary_path)
-
-	# Do nothing / pass-through
-	add_custom_target(process_base_module
-		COMMAND ${CMAKE_COMMAND} -E copy sof-pre sof-${fw_name}
-		DEPENDS prepare_sof_post_process
-	)
-endif()
+# Do nothing / pass-through
+add_custom_target(process_base_module
+	COMMAND ${CMAKE_COMMAND} -E copy sof-pre sof-${fw_name}
+	DEPENDS prepare_sof_post_process
+)
```

Comment:
I don't see the point of removing this part of the CMake code, it has absolutely nothing platform specific and we just don't know how companies outside Intel use rimage. I believe the point of this PR is to block everyone from building TGL+XTOS in this branch. This is already a large and significant change, please don't top it up with risky, unnecessary and **platform-independent** CMake code cleanups. Even less so considering the entire project has 0.6 CMake maintainers including myself.

Reply:
> nobody else uses the bootloader - check CONFIG_XT_BOOT_LOADER

Sure but what does removing this particular CMake code achieves? When the Linux kernel code removes support for old hardware it's neither for the purpose of saving disk space nor for aesthetics nor for "ideology". It's because old code gets in the way of updating APIs and testing changes. In other words Linux kernel code is removed _when it solves an actual, specific maintenance problem_. I don't see what specific problem is being solved by removing this CMake code here now.

I really want to minimize CMake churn as much as possible and considering I'm the only one (barely) maintaining sof/cmake code I think I deserve this. As a coincidence I've just performed some CMake archeology for #7546 and deleted CMake code there would have make that archeology even more painful that it already was. By the way we have many more rimage-related changes in flight right now.

Feel free to a "this is dead code now since..." comment, that would be useful.

Quoting from #7552 that I just submitted, #7552 and 34d7baa80061 similarly revert some massive search/replace that backfired:

> If we want to stop supporting something then let's keep things simple and remove it only where it actually gets in the way and where it actually becomes a problem. No need to search and scrub every reference to it from the face of the planet and accidentally make our life harder on release branches, create pointless git conflicts and increase maintenance costs when trying to reduce them!

Reply:
@marc-hb sorry, TBH I don't understand your objection here. We are removing the only use case for this block of code. If we keep it we'll keep stumbling over it trying to identify what it's there for, until someone removes it. Why - simple - to reduce maintenance costs. I might miss some further dependencies, breakage possibilities or whatever. But here it seems pretty clear to me - that `if` branch will never be entered. So it can be removed.

Reply:
>  Why - simple - to reduce maintenance costs.

I'm the only one doing CMake maintenance right now and I gave you examples. I'm sorry you don't understand them.

Reply:
> If we keep it we'll keep stumbling over it trying to identify what it's there for, until someone removes it. Why - simple - to reduce maintenance costs.

"we" has been only myself for the last few years since Janusz left. So please leave the estimation of maintenance costs to me and believe me when I say that I don't do it for fun.